examples/stm32/nucleo-f429zi-baremetal/main.c

```diff
@@ -31,20 +31,23 @@ void EXTI15_10_IRQHandler(void) {  // External interrupt handler
   gpio_write(LED1, gpio_read(BTN1));  // No debounce. Turn LED if button pressed
 }
 
-static void timer_cb(void *arg) {
+void SystemInit(void) {                  // Called automatically by startup code
+  clock_init();                          // Set clock to 180MHz
+  SysTick_Config(SYS_FREQUENCY / 1000);  // Increment s_ticks every ms
+}
```

Comment:
Yes... but the user might pull-in the provided system_devicename.c ... if this is going to be a skeleton we should perhaps write a system file or use it. Some set the clock to highest frequency, some don't...

Reply:
I think providing per-clock examples would be an overkill, TBH.
I do not expect any conflicts, if our examples are used as skeletons.
Conflicts can happen if user integrates our code into their existing example, which is a different use case, IMO.

Reply:
Oh, no, no, just the highest speed...
I meant keeping system stuff in a system file of our own or integrating with the vendor system file

Reply:
Perhaps just putting this into our system.c is the easiest/fastest. If the user wants to integrate with his example will know at first sight that there is system stuff to handle.
Probably the `clock_init()` function itself should move to that system file; when vendors do set the clock, they do it in the system_xxxx.c file